src/msg/async/AsyncConnection.cc

```diff
@@ -267,14 +267,16 @@ ssize_t AsyncConnection::read_until(unsigned len, char *p)
                                << " left is " << left << " buffer still has "
                                << recv_end - recv_start << dendl;
     if (left == 0) {
+      state_offset = 0;
       return 0;
     }
+
     state_offset += to_read;
   }
 
   recv_end = recv_start = 0;
   /* nothing left in the prefetch buffer */
-  if (len > recv_max_prefetch) {
```

Comment:
@yuyuyu101 This variable of left is more appropriate, because we have read part of data from the recv_buf.
if the len is greater than the recv_max_prefetch, the left maybe is less than recv_max_prefetch after reading recv_buf
so I think we should use left in this place.